File: backend/TravelApp.Api/Models/Activity.cs
namespace TravelApp.Api.Models;

public class Activity
{
    public int Id { get; set; }
    public int TravelPlanId { get; set; }
    public TravelPlan TravelPlan { get; set; } = null!;

    public DateTime DayDate { get; set; }

    public string Title { get; set; } = string.Empty;

    public string? Notes { get; set; }

    public string? Time { get; set; }

    public string? Location { get; set; }

    public decimal? Cost { get; set; }

    public ActivityStatus Status { get; set; } = ActivityStatus.Planned;
}

File: frontend/src/models/Activity.js
export const ACTIVITY_STATUSES = ['Planned', 'Reserved', 'Done', 'Cancelled'];

export function toActivity(data) {
  return {
    id: Number(data.id),
    dayDate: data.dayDate || '',
    title: data.title || '',
    notes: data.notes || null,
    time: data.time || null,
    location: data.location || null,
    cost: data.cost == null ? null : Number(data.cost),
    status: data.status || 'Planned',
  };
}
